app/containers/LoginLedgerNanoS/LoginLedgerNanoS.jsx

```diff
@@ -147,8 +274,17 @@ export default class LoginLedgerNanoS extends React.Component<Props, State> {
   }
 
   handleLogin = () => {
-    this.props.login(this.props.publicKey)
+    if (this.state.publicKeys.length && this.state.addressOption) {
+      const keyData = this.state.publicKeys.find(
+        publicKey =>
+          this.state.addressOption &&
```

Comment:
Flow was throwing an error (even though the check was already being performed on 27) will see if I can come up with a cleaner solution.

Reply:
Oh that's cause `this.state` might have changed since the check. Flow is a stickler :)
Just put the addressOption in a new variable and you're good to go:
```js
const { addressOption, publicKeys } = this.state
if (publicKeys.length && addressOption) {
  const keyData = publicKeys.find(
    publicKey => addressOption.value === publicKey.key
  )
  if (keyData) {
    this.props.login(keyData)
  }
}
```